crates/biome_fs/src/path.rs

```diff
@@ -50,7 +119,237 @@ impl BiomePath {
+        self.kind.0.contains(FileKindFlag::Config)
+    }
+
+    pub fn is_manifest(&self) -> bool {
+        self.kind.0.contains(FileKindFlag::Manifest)
+    }
+
+    pub fn is_to_inspect(&self) -> bool {
+        self.kind.0.contains(FileKindFlag::ToInspect)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl schemars::JsonSchema for FileKind {
+    fn schema_name() -> String {
+        String::from("FileKind")
+    }
+
+    fn json_schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        <Vec<FileKindFlag>>::json_schema(gen)
+    }
+}
+
+impl Deref for BiomePath {
+    type Target = PathBuf;
+
+    fn deref(&self) -> &Self::Target {
+        &self.path
+    }
+}
+
+impl PartialOrd for BiomePath {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for BiomePath {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let current_file_name = self.get_file_name();
+        let other_file_name = other.get_file_name();
+        match (current_file_name, other_file_name) {
+            (Some(current_file_name), Some(other_file_name)) => {
+                match Self::priority(current_file_name).cmp(&Self::priority(other_file_name)) {
+                    Ordering::Equal => self.path.cmp(&other.path),
+                    ordering => ordering,
+                }
+            }
+            (Some(_), None) => Ordering::Less,
+            (None, Some(_)) => Ordering::Greater,
+            (None, None) => Ordering::Equal,
+        }
+    }
+}
+
+impl From<&BiomePath> for BiomePath {
+    fn from(value: &BiomePath) -> Self {
+        BiomePath::new(value.path.clone())
+    }
+}
+
+#[derive(Debug, Clone, Hash, Default)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize, schemars::JsonSchema)
+)]
+/// It defines an alias for a path
+struct AliasPath {
+    /// The name given to the alias
+    name: String,
+    /// The path associated to `name`
+    prefix: Vec<PathBuf>,
+}
+
+impl AliasPath {
+    pub fn new(name: impl Into<String>) -> Self {
+        Self {
+            name: name.into(),
+            prefix: vec![],
+        }
+    }
+
+    pub fn add_prefix(&mut self, prefix: impl Into<PathBuf>) {
+        self.prefix.push(prefix.into());
+    }
+}
+
+pub struct Aliases {
+    aliases: Vec<AliasPath>,
+}
+
+impl Aliases {
+    pub fn new() -> Self {
+        Self { aliases: vec![] }
+    }
+
+    pub fn with_alias<'a>(
+        mut self,
+        name: impl Into<String>,
+        paths: impl Iterator<Item = &'a Path>,
+    ) -> Self {
+        let mut alias = AliasPath::new(name);
+        for path in paths {
+            alias.add_prefix(path);
+        }
+        self.aliases.push(alias);
+        self
+    }
+
+    pub fn get_paths_by_name(&self, name: &str) -> Option<&Vec<PathBuf>> {
+        self.aliases
+            .iter()
+            .find(|alias| alias.name == name)
+            .map(|alias| &alias.prefix)
+    }
```

Comment:
Do you think we could return a slice instead? This looks like an antipattern of returning `&Vec`.